webpack/scenes/RedHatRepositories/index.js

```diff
@@ -30,6 +31,14 @@ class RedHatRepositoriesPage extends Component {
     const { enabledRepositories, repositorySets } = this.props;
     const { repoParams } = createEnabledRepoParams(enabledRepositories);
 
+    if (repositorySets.missingPermissions && repositorySets.missingPermissions.length > 0) {
```

Comment:
Notice this mechanism:
```js
if (!havePermissions()) {
  return <PermissionDenied />
} else { continue }
```

It is also used in the `SubscriptionsPage` and we might adopt it to other pages.
Can it be reused as a HOC?
So your logic can be:
```js
if (!havePermissions()) {
  return <PermissionDenied />
} else { <Component /> }
```